libraries/ms-common/src/x86/gpio.c

```diff
@@ -12,7 +101,50 @@ static uint32_t prv_get_index(const GpioAddress *address) {
   return address->port * (uint32_t)NUM_GPIO_PORTS + address->pin;
 }
 
+static void prv_export() {
+  for (uint16_t i = 0; i < GPIO_TOTAL_PINS; i++) {
+    s_store.state[i] = s_pin_settings[i].state;
+  }
+  store_export(MX_STORE_TYPE__GPIO, &s_store, NULL);
+}
+
+static void update_store(ProtobufCBinaryData msg_buf, ProtobufCBinaryData mask_buf) {
+  MxGpioStore *msg = mx_gpio_store__unpack(NULL, msg_buf.len, msg_buf.data);
+  MxGpioStore *mask = mx_gpio_store__unpack(NULL, mask_buf.len, mask_buf.data);
+
+  for (uint16_t i = 0; i < mask->n_state; i++) {
+    // only update state if mask is set
+    if (mask->state[i] != 0) {
+      s_store.state[i] = msg->state[i];
+      if (s_pin_settings[i].state != (uint8_t)msg->state[i]) {
+        s_pin_settings[i].state = msg->state[i];
+        GpioAddress address = { .port = 0, .pin = i % 16 };
```

Comment:
shouldn't this be `.port = i / GPIO_PINS_PER_PORT, .pin = i % GPIO_PINS_PER_PORT`

Reply:
We have this unwritten requirement that there's only 16 interrupt slots, so we assign them based on pin number (not port). This is actually a huge issue and should be documented at some point. Added a comment here explaining this.